_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,354 @@
+  
+<br/>
+<br/>
+
+
+### Steps Involved
+
+All the scripts for cluster provisioning or deprovisioning are located at - [`${metal3-dev-env}/scripts/v1alphaX/`](https://github.com/metal3-io/metal3-dev-env/tree/master/scripts/v1alphaX). The scripts call a common playbook which handles all the tasks that are available.
+
+> info Note
+> *Here is a depiction of the common steps, mainly involving generating templates and config files.*  
+
+<br/>
+
+!["A diagram depicting the Generate Templates workflow"](/assets/2020-05-30-Metal3-Cluster-Deployment/metal3-generate-templates.png)
+<br/>
+<br/>
+
+The steps involved in the process are :
+
+- The script calls an ansible playbook with necessary parameter ( from env vars and defaults )
+- The playbook executes the role -, [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), which runs the corresponding [task_file](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/tasks) for provisioning/deprovisioning the cluster/controlplane or a worker
+- There are [templates](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/templates) in the role, which are copied to the `manifest` directory. They are used for generating configuration for the cluster and kubeadm, which is then supplied to the Kubernetes module of ansible to create the cluster.
+- During provisioning, first the `clusterctl` env file is generated, then and `manifest` file is created/updated using the `clusterctl` tool, which is then supplied to `kustomize` tool to create the config files for provisioning the resource, from the available templates.
+- The manifests are generated for - cluster, controlplane and machinedeployment, and the worker manifests are merged with the deployment manifests.
+- Finally a yaml file is generated with all the details for the cluster/controlplane/worker, which is then passed to K8s module in ansible which creates the resources.
+- There are also manifests stored in ${Manifests} directory - [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test/files/manifests`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), for the cluster which can be used to deprovision the cluster at a later point in time.
+> **Note** : * the `manifests` directory is not present but created when you first use the scripts.*
+- Centos or Ubuntu images [are downloaded](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/v1aX_integration_test/tasks/download_image.yml) when provisioning a controlplane or a worker  
+
+<br/>
+<br/>
+
+
+!["An Overview of various resources generated while provisioning and their relationship amongst themselves"](/assets/2020-05-30-Metal3-Cluster-Deployment/metal3-bmetal-arch-overview.png)  
+<br/>
+<br/>
+
+
+### Provision Cluster
```

Comment:
I'm wondering if it's probably worth creating a new L2 section here, before starting with the various `Provision` steps.  The section so far provided an overvew of the process. Here we begin to follow the actual process. For clarity I think it's better to separate both.

So, maybe here is where this would go:

```
## BareMetal Cluster Deployment

Now let's go ahead and start provisioning a cluster

### Provision Cluster
```

Then, in the section above you could replace its L2 header (which would now be here) with something like:

```
## metal3-dev-env cluster provisioning scripts overview
```

or something like that

Makes sense?